.github/workflows/pypi.yml

```diff
@@ -0,0 +1,52 @@
+name: Publish to PyPI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```

Comment:
@callumrollo let me know if you are OK renaming master to main. If not I can change this.